internal/db/nosql/cassandra/option_test.go

```diff
@@ -19,1026 +19,703 @@ package cassandra
-	       }
-	       return nil
-	   }
-	*/
+		if diff := comparator.Diff(obj, w.obj, clientComparatorOpts...); diff != "" {
+			return errors.New(diff)
+		}
+		return nil
+	}
 
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           hosts: nil,
-		       },
-		       want: want {
-		           obj: new(T),
-		       },
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           hosts: nil,
-		           },
-		           want: want {
-		               obj: new(T),
-		           },
-		       }
-		   }(),
-		*/
+		{
+			name: "set host success",
+			args: args{
+				hosts: []string{"vald.vdaas.org"},
+			},
+			want: want{
+				obj: &T{
+					hosts: []string{"vald.vdaas.org"},
+				},
+			},
+		},
+		{
+			name: "set host success if hosts is nil",
+			args: args{
+				hosts: nil,
+			},
+			want: want{
+				obj: &T{
+					hosts: nil,
+				},
+			},
+		},
+		{
+			name: "set host twice success",
+			args: args{
+				hosts: []string{"vald.vdaas.org"},
+			},
+			want: want{
+				obj: &T{
+					hosts: []string{"vald.vdaas.org"},
+				},
+			},
+			checkFunc: func(w want, obj *T, err error) error {
+				if err := defaultCheckFunc(w, obj, err); err != nil {
+					return err
+				}
+
+				// set host to the same obj again
+				if err := WithHosts("hosts1")(obj); err != nil {
+					return err
+				}
+
+				w = want{
+					obj: &T{
+						hosts: []string{"vald.vdaas.org", "hosts1"},
+					},
+				}
+				if err := defaultCheckFunc(w, obj, err); err != nil {
+					return err
+				}
+
+				return nil
+			},
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(tt *testing.T) {
-			defer goleak.VerifyNone(t)
+			defer goleak.VerifyNone(tt, goleakIgnoreOptions...)
 			if test.beforeFunc != nil {
 				test.beforeFunc(test.args)
 			}
 			if test.afterFunc != nil {
 				defer test.afterFunc(test.args)
 			}
 
-			// Uncomment this block if the option returns an error, otherwise delete it
-			/*
-			   if test.checkFunc == nil {
-			       test.checkFunc = defaultCheckFunc
-			   }
-
-			   got := WithHosts(test.args.hosts...)
-			   obj := new(T)
-			   if err := test.checkFunc(test.want, obj, got(obj)); err != nil {
-			       tt.Errorf("error = %v", err)
-			   }
-			*/
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
 
-			// Uncomment this block if the option returns an error, otherwise delete it
-			/*
-			   if test.checkFunc == nil {
-			       test.checkFunc = defaultCheckFunc
-			   }
-			   got := WithHosts(test.args.hosts...)
-			   obj := new(T)
-			   got(obj)
-			   if err := test.checkFunc(tt.want, obj); err != nil {
-			       tt.Errorf("error = %v", err)
-			   }
-			*/
+			got := WithHosts(test.args.hosts...)
+			obj := new(T)
+			if err := test.checkFunc(test.want, obj, got(obj)); err != nil {
```

Comment:
```suggestion
			obj := new(T)
			got(obj)
			if err := test.checkFunc(test.want, obj, got); err != nil {
```
No need to change template code more than necessary

Reply:
It is from template code for `option that return error` case. (line 133)

Reply:
oh, sorry